pkg/e2e/actuators/actuators.go

```diff
@@ -0,0 +1,411 @@
+package actuators
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	e2e "github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/framework"
+	mapiv1beta1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	kpolicyapi "k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	workerRoleLabel = "node-role.kubernetes.io/worker"
+	waitShort       = 1 * time.Minute
+	waitMedium      = 3 * time.Minute
+	waitLong        = 10 * time.Minute
+	retryMedium     = 5 * time.Second
+)
+
+var nodeDrainLabels = map[string]string{
+	workerRoleLabel:      "",
+	"node-draining-test": string(uuid.NewUUID()),
+}
+
+func machineFromMachineset(machineset *mapiv1beta1.MachineSet) *mapiv1beta1.Machine {
+	randomUUID := string(uuid.NewUUID())
+
+	machine := &mapiv1beta1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: machineset.Namespace,
+			Name:      "machine-" + randomUUID[:6],
+			Labels:    machineset.Labels,
+		},
+		Spec: machineset.Spec.Template.Spec,
+	}
+	if machine.Spec.ObjectMeta.Labels == nil {
+		machine.Spec.ObjectMeta.Labels = map[string]string{}
+	}
+	for key := range nodeDrainLabels {
+		if _, exists := machine.Spec.ObjectMeta.Labels[key]; exists {
+			continue
+		}
+		machine.Spec.ObjectMeta.Labels[key] = nodeDrainLabels[key]
+	}
+	return machine
+}
+
+func isNodeReady(node *corev1.Node) bool {
+	for _, c := range node.Status.Conditions {
+		if c.Type == corev1.NodeReady {
+			return c.Status == corev1.ConditionTrue
+		}
+	}
+	return false
+}
+
+func waitUntilNodesAreReady(client runtimeclient.Client, listOpt *runtimeclient.ListOptions, nodeCount int) error {
+	return wait.PollImmediate(retryMedium, waitLong, func() (bool, error) {
+		nodes := corev1.NodeList{}
+		if err := client.List(context.TODO(), listOpt, &nodes); err != nil {
+			glog.Errorf("Error querying api for Node object: %v, retrying...", err)
+			return false, nil
+		}
+		// expecting nodeGroupSize nodes
+		nodeCounter := 0
+		for _, node := range nodes.Items {
+			if _, exists := node.Labels[workerRoleLabel]; !exists {
+				continue
+			}
+
+			if !isNodeReady(&node) {
+				continue
+			}
+
+			nodeCounter++
+		}
+
+		if nodeCounter < nodeCount {
+			glog.Errorf("Expecting %v nodes with %#v labels in Ready state, got %v", nodeCount, nodeDrainLabels, nodeCounter)
+			return false, nil
+		}
+
+		glog.Infof("Expected number (%v) of nodes with %v label in Ready state found", nodeCount, nodeDrainLabels)
```

Comment:
Shouldn't `nodeCount` be `nodeCounter`?

Reply:
`nodeCount` is the expected number (it could be even more in real though it needs to be at least `nodeCount`)